src/SampleFormat.h

```diff
@@ -125,18 +129,29 @@ class GrowableSampleBuffer : private SampleBuffer
 //
 
 AUDACITY_DLL_API
-void      CopySamples(constSamplePtr src, sampleFormat srcFormat,
-                      samplePtr dst, sampleFormat dstFormat,
-                      unsigned int len, bool highQuality=true,
-                      unsigned int srcStride=1,
-                      unsigned int dstStride=1);
+//! Copy samples from any format into the widest format, which is 32 bit float, with no dithering
+/*!
+ @param src address of source samples
+ @param srcFormat format of source samples, determines sizeof each one
+ @param dst address of floating-point numbers
+ @param len count of samples to copy
+ @param srcStride how many samples to advance src after copying each one
+ @param dstString how many samples to advance dst after copying each one
+ */
+void SamplesToFloats(constSamplePtr src, sampleFormat srcFormat,
+    float *dst, size_t len, size_t srcStride = 1, size_t dstStride = 1);
 
 AUDACITY_DLL_API
-void      CopySamplesNoDither(samplePtr src, sampleFormat srcFormat,
-                      samplePtr dst, sampleFormat dstFormat,
-                      unsigned int len,
-                      unsigned int srcStride=1,
-                      unsigned int dstStride=1);
+//! Copy samples from any format to any other format; apply dithering only if narrowing the format
+/*!
+ @copydetails SamplesToFloats()
+ @param dstFormat format of destination samples, determines sizeof each one
+ @param ditherType choice of dithering algorithm to use if narrowing the format
+ */
+void CopySamples(constSamplePtr src, sampleFormat srcFormat,
+   samplePtr dst, sampleFormat dstFormat, size_t len,
+   DitherType ditherType = gHighQualityDither,
```

Comment:
This is the first time I've seen a default parameter in a function that is not a compile time constant.  A comment explaining when it is evaluated is called for.